tiledb/api/c_api/attribute/attribute_api.cc

```diff
@@ -135,9 +135,7 @@ int32_t tiledb_attribute_get_cell_size(
 int32_t tiledb_attribute_dump(
     const tiledb_attribute_handle_t* attr, FILE* out) {
   ensure_attribute_is_valid(attr);
-
-  if (out == nullptr)
-    return TILEDB_ERR;
+  ensure_output_pointer_is_valid(out);
```

Comment:
This isn't the right function to call because it generates the wrong error. Output argument are pointers `T *x` where we assign to the argument `*x = y`. `out` is not an output argument in this sense.

When a `FILE *` is null, it needs a separate error message. Since this error occurs in multiple places, it would good to put it into `tiledb/api/c_api_support/argument_validation.h`. `ensure_cstream_handle_is_valid` would be a good name.

Reply:
this is already used like this: https://github.com/TileDB-Inc/TileDB/blob/agis/add-dump-functions-with-str-argument/tiledb/sm/c_api/tiledb.cc#L241